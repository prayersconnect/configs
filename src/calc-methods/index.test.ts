import {
  getCalcMethodByName,
  getCalculationMethods,
  getCalculationMethodByCountry,
  getJuristicMethods,
} from './index';

describe('calc-methods', () => {
  describe('getCalcMethodByName', () => {
    it('returns undefined if no method found', () => {
      // @ts-expect-error invalid type
      const method = getCalcMethodByName('foo');
      expect(method).toBeUndefined();
    });

    it('returns method if found', () => {
      const method = getCalcMethodByName('Egyptian');
      expect(method).toMatchSnapshot();
    });
  });

  describe('getCalculationMethods', () => {
    it('returns all methods', () => {
      const methods = getCalculationMethods();
      expect(methods).toMatchSnapshot();
      methods.forEach((method) => {
        expect(method).toHaveProperty('calculationKey');
        expect(method.calculationKey).toMatchSnapshot(
          `values for ${method.calculationKey}`
        );
      });
    });
  });

  describe('getCalcMethodsByCountry', () => {
    it('returns method and asr method for USA', () => {
      const methods = getCalculationMethodByCountry('United States');
      expect(methods).toMatchInlineSnapshot(`
        {
          "asrCalculation": "Standard",
          "calculationMethod": "IslamicSocietyOfNorthAmerica",
        }
      `);
    });
    it('returns method and asr method for Turkey', () => {
      const methods = getCalculationMethodByCountry('Turkey');
      expect(methods).toMatchInlineSnapshot(`
        {
          "asrCalculation": "Standard",
          "calculationMethod": "Turkey",
        }
      `);
    });
    it('returns method and asr method for UK', () => {
      const methods = getCalculationMethodByCountry('United Kingdom');
      expect(methods).toMatchInlineSnapshot(`
        {
          "asrCalculation": "Standard",
          "calculationMethod": "MoonsightingCommittee",
        }
      `);
    });
    it('returns method and asr method for AU', () => {
      const methods = getCalculationMethodByCountry('Australia');
      expect(methods).toMatchInlineSnapshot(`
        {
          "asrCalculation": "Hanafi",
          "calculationMethod": "MuslimWorldLeague",
        }
      `);
    });
    it('returns method and asr method for BD', () => {
      const methods = getCalculationMethodByCountry('Bangladesh');
      expect(methods).toMatchInlineSnapshot(`
        {
          "asrCalculation": "Hanafi",
          "calculationMethod": "Karachi",
        }
      `);
    });
    it('returns method and asr method for Russia', () => {
      const methods = getCalculationMethodByCountry('Russia');

      expect(methods).toMatchInlineSnapshot(`
        {
          "asrCalculation": "Standard",
          "calculationMethod": "Russia",
        }
      `);
    });
    it('returns default config if country not found', () => {
      const methods = getCalculationMethodByCountry('foo');
      expect(methods).toMatchInlineSnapshot(`
        {
          "asrCalculation": "Standard",
          "calculationMethod": "MuslimWorldLeague",
        }
      `);
    });
    it('returns null if country not provided', () => {
      const methods = getCalculationMethodByCountry(null as any);
      expect(methods).toBeNull();
    });
  });

  describe('getJuristicMethods', () => {
    it('returns all asr methods', () => {
      const methods = getJuristicMethods();
      expect(methods).toMatchSnapshot();
    });
  });
});
