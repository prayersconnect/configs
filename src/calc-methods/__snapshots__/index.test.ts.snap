// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calc-methods getCalcMethodByName returns method if found 1`] = `
{
  "calculationKey": "Egyptian",
  "get": [Function],
  "info": "Fajr: 19.5°, Isha: 17.5°",
  "label": "Egyptian General Survey Authority",
  "region": "Africa, Syria, Iraq, Lebanon, Malaysia, Parts of the USA",
}
`;

exports[`calc-methods getCalculationMethods returns all methods 1`] = `
[
  {
    "calculationKey": "Custom",
    "get": [Function],
    "info": "Fajr: 0°, Isha: 0°",
    "label": "Custom",
  },
  {
    "calculationKey": "Algeria",
    "get": [Function],
    "info": "Fajr: 18°, Magrib: +3 min, Isha: 17°",
    "label": "Ministry of Religious Affairs and Wakfs, Algeria",
    "otherLabel": "Ministry of Religious Affairs and Wakfs",
  },
  {
    "calculationKey": "Brunei",
    "get": [Function],
    "info": "Fajr: 20°, Isha: 18°",
    "label": "Kementrian Hal Ehwal Ugama (Brunei Darussalam)",
    "otherLabel": "Brunei Darussalam",
  },
  {
    "calculationKey": "Dubai",
    "get": [Function],
    "info": "Fajr: 18.2°, Isha: 18.2°",
    "label": "The Gulf Region (Dubai)",
    "url": "https://www.awqaf.gov.ae/en/Pages/PrayerTimes.aspx",
  },
  {
    "calculationKey": "Egyptian",
    "get": [Function],
    "info": "Fajr: 19.5°, Isha: 17.5°",
    "label": "Egyptian General Survey Authority",
    "region": "Africa, Syria, Iraq, Lebanon, Malaysia, Parts of the USA",
  },
  {
    "calculationKey": "France",
    "get": [Function],
    "info": "Fajr: 12.0°, Isha: 12.0°",
    "label": "Union Organization Islamic de France - 12",
    "region": "France region",
  },
  {
    "calculationKey": "France15",
    "get": [Function],
    "info": "Fajr: 15.0°, Isha: 15.0°",
    "label": "Union Organization Islamic de France - 15",
    "region": "France region",
  },
  {
    "calculationKey": "France18",
    "get": [Function],
    "info": "Fajr: 18.0°, Isha: 18.0°",
    "label": "Union Organization Islamic de France - 18",
    "region": "France region",
  },
  {
    "calculationKey": "Gulf",
    "get": [Function],
    "info": "Fajr: 19.5°, Isha: +90 min",
    "label": "Gulf region",
  },
  {
    "calculationKey": "Jafari",
    "get": [Function],
    "info": "Fajr: 16.0°, Magrib: 4.0°, Isha: 14.0°",
    "label": "Shia Ithna Ashari, Leva Institute, Qum",
    "otherLabel": "Leva Institute, Qum",
  },
  {
    "calculationKey": "Karachi",
    "get": [Function],
    "info": "Fajr: 18.0°, Isha: 18.0°",
    "label": "University of Islamic Sciences, Karachi",
    "otherLabel": "University of Islamic Sciences",
    "region": "Pakistan, Bangladesh, India, Afghanistan, Parts of Europe",
  },
  {
    "calculationKey": "Indonesia",
    "get": [Function],
    "info": "Fajr: 20.0°, Isha: 18.0°",
    "label": "Kementrian Agama Republik Indonesia (KEMENAG)",
    "otherLabel": "Kementrian Agama Republik Indonesia",
  },
  {
    "calculationKey": "Kuwait",
    "get": [Function],
    "info": "Fajr: 18.0°, Isha: 17.5°",
    "label": "Kuwait",
  },
  {
    "calculationKey": "MoonsightingCommittee",
    "get": [Function],
    "info": "Fajr: 18.0°, Isha: 18.0°, Note: seasonal adjustment values.",
    "label": "Moonsighting Committee",
    "region": "UK",
  },
  {
    "calculationKey": "MuslimWorldLeague",
    "get": [Function],
    "info": "Fajr: 18.0°, Isha: 17.0°",
    "label": "Muslim World League",
    "region": "Europe, The Far East, Parts of the USA",
  },
  {
    "calculationKey": "IslamicSocietyOfNorthAmerica",
    "get": [Function],
    "info": "Fajr: 15.0°, Isha: 15.0°",
    "label": "Islamic Society of North America - ISNA",
    "region": "Parts of the USA, Canada, Parts of the UK",
  },
  {
    "calculationKey": "Qatar",
    "get": [Function],
    "info": "Fajr: 18.5°, Isha: +90 min",
    "label": "Qatar",
    "url": "https://www.qatarch.com/home",
  },
  {
    "calculationKey": "Russia",
    "get": [Function],
    "info": "Fajr: 16.0°, Isha: 15.0°",
    "label": "Spiritual Administration of Muslims of Russia",
  },
  {
    "calculationKey": "Singapore",
    "get": [Function],
    "info": "Fajr: 20.0°, Isha: 18.0°",
    "label": "Singapore",
    "region": "Singapore region",
    "url": "https://www.muis.gov.sg/",
  },
  {
    "calculationKey": "Tehran",
    "get": [Function],
    "info": "Fajr: 17.7°, Magrib: 4.5°, Isha: 14.0°",
    "label": "Shia, Institute of Geophysics, University of Tehran",
  },
  {
    "calculationKey": "Tunisia",
    "get": [Function],
    "info": "Fajr: 18.0°, Isha: 18.0°",
    "label": "Ministry of Religious Affairs of Tunisia",
  },
  {
    "calculationKey": "Turkey",
    "get": [Function],
    "info": "Fajr: 18.0°, Isha: 17.0°",
    "label": "Presidency of Religious Affairs, Turkey",
    "postAdjustments": [Function],
    "preAdjustments": [Function],
    "url": "https://kurul.diyanet.gov.tr/Karar-Mutalaa-Cevap/4093/45-enlemin-otesinde-namaz-vakitleri",
  },
  {
    "calculationKey": "UmmAlQura",
    "get": [Function],
    "info": "Fajr: 18.5°, Isha: +90 min, Ramadan - Isha: +120 min",
    "label": "Umm al-Qura University, Makkah",
    "preAdjustments": [Function],
    "region": "The Arabian Peninsula",
    "url": "https://www.ummulqura.org.sa/Index.aspx",
  },
]
`;

exports[`calc-methods getCalculationMethods returns all methods: values for Algeria 1`] = `"Algeria"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Brunei 1`] = `"Brunei"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Custom 1`] = `"Custom"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Dubai 1`] = `"Dubai"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Egyptian 1`] = `"Egyptian"`;

exports[`calc-methods getCalculationMethods returns all methods: values for France 1`] = `"France"`;

exports[`calc-methods getCalculationMethods returns all methods: values for France15 1`] = `"France15"`;

exports[`calc-methods getCalculationMethods returns all methods: values for France18 1`] = `"France18"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Gulf 1`] = `"Gulf"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Indonesia 1`] = `"Indonesia"`;

exports[`calc-methods getCalculationMethods returns all methods: values for IslamicSocietyOfNorthAmerica 1`] = `"IslamicSocietyOfNorthAmerica"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Jafari 1`] = `"Jafari"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Karachi 1`] = `"Karachi"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Kuwait 1`] = `"Kuwait"`;

exports[`calc-methods getCalculationMethods returns all methods: values for MoonsightingCommittee 1`] = `"MoonsightingCommittee"`;

exports[`calc-methods getCalculationMethods returns all methods: values for MuslimWorldLeague 1`] = `"MuslimWorldLeague"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Qatar 1`] = `"Qatar"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Russia 1`] = `"Russia"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Singapore 1`] = `"Singapore"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Tehran 1`] = `"Tehran"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Tunisia 1`] = `"Tunisia"`;

exports[`calc-methods getCalculationMethods returns all methods: values for Turkey 1`] = `"Turkey"`;

exports[`calc-methods getCalculationMethods returns all methods: values for UmmAlQura 1`] = `"UmmAlQura"`;

exports[`calc-methods getJuristicMethods returns all asr methods 1`] = `
[
  {
    "label": "Shafii, Maliki, Jafari & Hanbali",
    "name": "Standard",
  },
  {
    "label": "Hanafi",
    "name": "Hanafi",
  },
]
`;
